src/plugins/fts/fts-build-mail.c

```diff
@@ -422,6 +422,7 @@ static int fts_build_data(struct fts_mail_build_context *ctx,
 		    FTS_BACKEND_FLAG_BUILD_FULL_WORDS) != 0) {
 		return fts_build_full_words(ctx, data, size, last);
 	} else {
+		if((!last) && (ctx->body_parser != NULL)) return 0;
 		return fts_backend_update_build_more(ctx->update_ctx, data, size);
```

Comment:
I think this is leaving out a lot more than just binary data.

Reply:
testing on production server makes it consistent (attachments are indexed, and others field also as usual).

Do you have a use case of not working situation ?

Reply:
To make it proper and exact, the "fts_body_parser_finish" shall be indeed called first, before threst

Reply:
I think what this change does is that it indexes only the last piece of the body parts, no matter what the body part's contents are. So if it's a large blob of binary garbage, it probably indexes the last few (8?) kilobytes of the garbage. But also if it's a large text block it also indexes only the ending of the text.

I am interested though in what kind of emails cause this binary output? FTS is supposed to be indexing only MIME parts that have text/* content-type. Are some such emails wrongly containing binary in its text parts? Or is there something else going wrong? I've also heard that Tika can output garbage from some attachments.

Reply:
No, this is the indexing of attachment, getting out of fts_decoder option.

the indexer (xapian, squat, etc...) receives twice the attachment : first a binary , then the converted data .

The second part shall be the only one

that second part is sent only in the "finish" function